core/src/main/java/io/stargate/core/CoreActivator.java

```diff
@@ -2,26 +2,45 @@
 
 import com.codahale.metrics.health.HealthCheckRegistry;
 import io.stargate.core.activator.BaseActivator;
+import io.stargate.core.metrics.api.HttpMetricsTagProvider;
 import io.stargate.core.metrics.api.Metrics;
 import io.stargate.core.metrics.api.MetricsScraper;
 import io.stargate.core.metrics.impl.MetricsImpl;
-import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 public class CoreActivator extends BaseActivator {
 
+  /**
+   * Id if the {@link io.stargate.core.metrics.api.HttpMetricsTagProvider}. If not set, this
+   * activator will register a default impl.
+   */
+  private static final String HTTP_TAG_PROVIDER_ID =
+      System.getProperty("stargate.metrics.http_tag_provider.id");
+
   public CoreActivator() {
     super("core services");
   }
 
   @Override
   protected List<ServiceAndProperties> createServices() {
     MetricsImpl metricsImpl = new MetricsImpl();
-    return Arrays.asList(
-        new ServiceAndProperties(metricsImpl, Metrics.class, null),
-        new ServiceAndProperties(metricsImpl, MetricsScraper.class, null),
+
+    List<ServiceAndProperties> services = new ArrayList<>();
+    services.add(new ServiceAndProperties(metricsImpl, Metrics.class, null));
+    services.add(new ServiceAndProperties(metricsImpl, MetricsScraper.class, null));
+    services.add(
         new ServiceAndProperties(new HealthCheckRegistry(), HealthCheckRegistry.class, null));
+
+    // register default http tag provider if we are not using any special one
+    if (null == HTTP_TAG_PROVIDER_ID) {
```

Comment:
instead of this if, you may consider adding a default for ` System.getProperty("stargate.metrics.http_tag_provider.id");`:
```
 System.getProperty("stargate.metrics.http_tag_provider.id", "default_id");
```

Reply:
@tomekl007 The idea here is that if the property is not set, then core will initialize the default.. If we go for the default id, then we need to always specify this option? Or what exactly would be the idea here, can you help plz?

Reply:
Ok, so what will happen if `HTTP_TAG_PROVIDER_ID != null`? Where it will be initialized?

Reply:
OK, but this is the same as if I would ask if `HTTP_TAG_PROVIDER_ID == someValueMadeUp` who would set it then? Nobody and start would fail.. 

The whole idea is to set a value that is valid and then the specific module can react on that known/valid value by registering a special impl. The core would back-off then. Again I think my solution is simpler as you don't have to specify anywhere this option by default and core would registered a default one.

Reply:
Ok, so if any other module provides its own `HttpMetricsTagProvider`, it means that it should set the `stargate.metrics.http_tag_provider.id`, right?

Reply:
True, or this can be set from the outside I hope.. I am quite not into the modules and how these props are set internally and shared..